### ISSUE TITLE
Instruction Parsing Problems

### ISSUE THREAD

**Issue by Kneesnap** [beardypig/ghidra-emotionengine#38]
It seems at least one instruction is not parsed properly by Ghidra.
I have built a .S file. When I open the resulting ELF file in Ghidra, it misidentifies some of the instructions.

The .S file:
![image](https://user-images.githubusercontent.com/6562913/103393615-62729580-4ad8-11eb-8bc0-63517ad5638f.png)

Ghidra
![image](https://user-images.githubusercontent.com/6562913/103393557-0740a300-4ad8-11eb-816d-520bba72faf2.png)

NOTE: The reason the .S file looks funky is because it's auto-generated.

**Comment by Kneesnap** [comment]
Another Instruction:
![image](https://user-images.githubusercontent.com/6562913/103394737-27c02b80-4adf-11eb-802e-eb8474343ef2.png)
![image](https://user-images.githubusercontent.com/6562913/103394725-1840e280-4adf-11eb-9346-69a2be814596.png)

**Comment by astrelsky** [comment]
These are all correct. They are simplified. For example `daddu v0, zero, zero` is `v0 = 0 + 0` which is the same as `dmove v0, 0` or
`v0 = 0`

**Comment by Kneesnap** [comment]
dmove is not a valid instruction in the emotion engine though. That's the problem.

**Comment by astrelsky** [comment]
![Capture](https://user-images.githubusercontent.com/46897303/103442019-585bae80-4c20-11eb-88b4-fee23380ebf9.PNG)

**Comment by Kneesnap** [comment]
Just because it shows up in an emulator doesn't mean it's valid. Neither the official PS2 instruction set documentation, nor the PS2SDK recognize either `qmove` or `dmove` as valid instructions.

![image](https://user-images.githubusercontent.com/6562913/103450054-b6f74b80-4c65-11eb-9501-8bbaae85ddf1.png)

**Comment by astrelsky** [comment]
@beardypig

**Comment by asmblur** [comment]
I think it would be nice to have an option to disable those macros.

**Comment by beardypig** [comment]
I don't think it would be very easy to add an option to enable/disable these instructions because of how sleigh works, but I might be wrong.
They were borrowed from other instruction sets to simplify the decompilation results, similar to what pcsx2 has in their debugger. I never intended for the decompiled output from this extension to be reused, but if people will find it more useful to use the strict instruction set then it can be changed.

We could remove them as they may no longer be required - but, my guess is that it would be a breaking change and might break existing projects.

Edit: upon further investigation, we could make this pseudo instructions optional using a flag.

**Comment by Mc-muffin** [comment]
> Just because it shows up in an emulator doesn't mean it's valid. Neither the official PS2 instruction set documentation, nor the PS2SDK recognize either `qmove` or `dmove` as valid instructions.
> 
> ![image](https://user-images.githubusercontent.com/6562913/103450054-b6f74b80-4c65-11eb-9501-8bbaae85ddf1.png)

Just as a reference, PCSX2's debugger was implemented by the same person that made armips, and as such it has some macros defined, but you can still view the un-macro'd instrunctions (and the bytes they are decoded from) by pressing tab in the assembly view

**Comment by astrelsky** [comment]
> I don't think it would be very easy to add an option to enable/disable these instructions because of how sleigh works, but I might be wrong.
> They were borrowed from other instruction sets to simplify the decompilation results, similar to what pcsx2 has in their debugger. I never intended for the decompiled output from this extension to be reused, but if people will find it more useful to use the strict instruction set then it can be changed.
> 
> We could remove them as they may no longer be required - but, my guess is that it would be a breaking change and might break existing projects.
> 
> Edit: upon further investigation, we could make this pseudo instructions optional using a flag.

We could. Not sure if we mean the same thing, but we could also use the context register. You would need to redisassemble to see the changes though.
